pkg/serverless/daemon/routes.go

```diff
@@ -86,6 +89,7 @@ type EndInvocation struct {
 
 func (e *EndInvocation) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	log.Debug("Hit on the serverless.EndInvocation route.")
+	e.daemon.SetExecutionSpanComplete(true)
```

Comment:
I'd suggest doing this as a defer instead. Or at least right right after we actually complete the span.

Reply:
Hmm, actually, this takes some careful thought. I think this is actually the correct thing to do. No defer.

If end-invocation has been called, even if it is not done, we do not want to also call it a second time on impending timeout. That could potentially lead to two invocation spans.